---
title: "Small data example"
author: "William R.P. Denault"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Small data example}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(
  comment = "#",
  collapse = TRUE,
  results  = "markup",
  fig.align = "center",
  fig.width = 4.5,
  fig.height = 3.2,
  warning = FALSE,
  message = FALSE
)
set.seed(1)

```

This is vignette is a modified example based on Figure 1 panel B-C-D in [Denault et al paper](https://doi.org/10.1101/2025.05.16.654543).


To reproduce, install the latest development version of susieR (optional):
```{r eval=FALSE }
# install.packages("remotes")
remotes::install_github("stephenslab/susieR")
```

## Data
We analyze a simulated eQTL dataset by modifying normalized gene expression and normalized SNP genotypes in [Denault et al](https://doi.org/10.1101/2025.05.16.654543). The goal is to fine-map causal variants for expression (eQTLs). In this example the ground truth is unknown.

```{r load-data}
library(susieR)
data(data_small)
y <- data_small$y
X <- data_small$X
dim(X)
```

## Baseline SuSiE fit
The original SuSiE procedure displays signs of misscalibration: the
result is highly suspicious as we find 10 credible sets in a data set
containing only 47 samples.

```{r run-susie, fig.height=5, fig.width=5, message=FALSE}
res_susie <- susie(X,y,L = 10,verbose = FALSE)
res_susie$sets$cs
susie_plot(res_susie,y = "PIP")
```

Another clue is that the fine-mapped variants explain >99% of the
variation in gene expression, which suggests overfitting.

```{r, fig.height=5.5, fig.width=5}
ypred <- predict(res_susie, X)
pve   <- 1 - drop(res_susie$sigma2 / var(y))
round(100 * pve, 3)

plot(y, ypred, pch = 20,
     xlab = "Phenotype",
     ylab = "Predicted")
abline(0, 1, col = "magenta", lty = "dotted")
```

## SuSiE with Servin–Stephens SER (small = TRUE)
Setting `small = TRUE` switches SuSiE to a variant of single-effect regression (SER) that properly accounts for uncertainty in the residual, i.e. Servin-Stephens single effect regression ([Servin and Stephens 2007 PLOS Genetics](https://doi.org/10.1371/journal.pgen.0030114)).

```{r run-susie-small, message=FALSE, warning=FALSE} 
res_susie_small <- susie(X,y,L = 10,small = TRUE,verbose = TRUE)
res_susie_small$sets$cs
```

The results are more plausible as this version of SuSiE only a single CS.

```{r }
susie_plot(res_susie_small,y = "PIP")
```

The lead SNP in this CS explains about 51% of the variance
in expression, which is again more plausible

```{r}
i <- res_susie_small$sets$cs$L1
round(cor(X[,i]),digits = 3)
i <- i[1]
fit = lm(y ~ X[,i])
pred_small = fit$fitted.values
pve <-summary( fit)$r.squared
round(100*pve,digits = 3)
```
 
 
## Compare predictions: overfit (default) vs Servin–Stephens SER
 
```{r, fig.height=5, fig.width=5}
 
plot(y, ypred, pch = 20,
     xlab = "Phenotype",
     ylab = "Predicted")
points(y, pred_small, pch = 20, col = "blue")
abline(0, 1, col = "magenta", lty = "dotted")
legend("topleft", pch = c(20, 20), col = c("black", "blue"),
       legend = c("SuSiE (default Gaussian SER)",
                  "SuSiE (Servin–Stephens SER)") )
```
 
 
**References**

Servin, B. & Stephens, M. (2007). *Imputation-based analysis of association studies: Candidate regions and quantitative traits.* PLOS Genetics, 3(7): e114.


Denault et al. (2025) *Accounting for uncertainty in residual variances improves calibration for fine-mapping with small sample sizes*, bioRxiv
