<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Fine-mapping with summary statistics • susieR</title>
<!-- katex math --><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.16.11/dist/katex.min.css" integrity="sha384-nB0miv6/jRmo5UMMR1wu3Gz6NLsoTkbqJghGIsx//Rlm+ZU03BU6SQNC66uf4l5+" crossorigin="anonymous">
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.11/dist/katex.min.js" integrity="sha384-7zkQWkzuo3B5mTepMUcHkMB5jZaolc2xDwL6VFqjFALcbeS9Ggm/Yr2r3Dy4lfFg" crossorigin="anonymous"></script><script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.11/dist/contrib/auto-render.min.js" integrity="sha384-43gviWU0YVjaDtb/GhzOouOXtZMP/7XUzwPTstBeZFe/+rCMvRwr4yROQP43s0Xk" crossorigin="anonymous" onload="renderMathInElement(document.body);"></script><script src="../katex-auto.js"></script><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Roboto-0.4.10/font.css" rel="stylesheet">
<link href="../deps/JetBrains_Mono-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Roboto_Slab-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Fine-mapping with summary statistics">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">susieR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.14.21</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/finemapping.html" aria-label="Get started with susieR"><span class="fa fa-play-circle"></span> Get Started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html" aria-label="Function reference"><span class="fa fa-file-code"></span> Functions</a></li>
<li class="active nav-item"><a class="nav-link" href="../articles/index.html" aria-label="Articles and vignettes"><span class="fa fa-book-reader"></span> Vignettes</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html" aria-label="Package news"><span class="fa fa-newspaper"></span> News</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/stephenslab/susieR" aria-label="View source on GitHub"><span class="fa fab fa-github"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Fine-mapping with summary statistics</h1>
                        <h4 data-toc-skip class="author">Yuxin Zou and
Gao Wang</h4>
            
            <h4 data-toc-skip class="date">2025-09-24</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/stephenslab/susieR/blob/HEAD/vignettes/finemapping_summary_statistics.Rmd" class="external-link"><code>vignettes/finemapping_summary_statistics.Rmd</code></a></small>
      <div class="d-none name"><code>finemapping_summary_statistics.Rmd</code></div>
    </div>

    
    
<p>This vignette demonstrates how to use <code>susieR</code> with
“summary statistics” in the context of genetic fine-mapping. We use the
same simulated data as in <a href="finemapping.html">fine mapping
vignette</a>. The simulated data are expression levels of a gene (<span class="math inline">y</span>) in <span class="math inline">N \approx
600</span> individuals. We want to identify with the genotype matrix
<span class="math inline">X_{N\times
P}</span> (<span class="math inline">P=1001</span>) the genetic
variables that causes changes in expression level. This data set is
shipped with <code>susieR</code>. It is simulated to have exactly three
non-zero effects.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/stephenslab/susieR" class="external-link">susieR</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="the-data-set">The data-set<a class="anchor" aria-label="anchor" href="#the-data-set"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">N3finemapping</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/attach.html" class="external-link">attach</a></span><span class="op">(</span><span class="va">N3finemapping</span><span class="op">)</span></span>
<span><span class="va">n</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">X</span><span class="op">)</span></span></code></pre></div>
<p>Notice that we’ve simulated two sets of <span class="math inline">Y</span> as two simulation replicates. Here we’ll
focus on the first data-set.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">Y</span><span class="op">)</span></span>
<span><span class="co"># [1] 574   2</span></span></code></pre></div>
<p>Here are the three true signals in the first data-set:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">b</span> <span class="op">&lt;-</span> <span class="va">true_coef</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">b</span>, pch<span class="op">=</span><span class="fl">16</span>, ylab<span class="op">=</span><span class="st">'effect size'</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-4-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">b</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="co"># [1] 403 653 773</span></span></code></pre></div>
<p>So the underlying causal variables are 403, 653 and 773.</p>
</div>
<div class="section level2">
<h2 id="summary-statistics-from-simple-regression">Summary statistics from simple regression<a class="anchor" aria-label="anchor" href="#summary-statistics-from-simple-regression"></a>
</h2>
<p>Summary statistics of genetic association studies typically contain
effect sizes (<span class="math inline">\hat{\beta}</span> coefficient
from regression) and p-values. These statisticscan be used to perform
fine-mapping with given an additional input of correlation matrix
between variables. The correlation matrix in genetics is typically
referred to as an “LD matrix” (LD is short for linkage disequilibrium).
One may use external reference panels to estimate it when this matrix
cannot be obtained from samples directly. <em>Importantly, the LD matrix
has to be a matrix containing estimates of the correlation, <span class="math inline">r</span>, and not <span class="math inline">r^2</span> or <span class="math inline">|r|</span>.</em> See also <a href="susierss_diagnostic.html">this vignette</a> for how to check the
consistency of the LD matrix with the summary statistics.</p>
<p>The <code>univariate_regression</code> function can be used to
compute summary statistics by fitting univariate simple regression
variable by variable. The results are <span class="math inline">\hat{\beta}</span> and <span class="math inline">SE(\hat{\beta})</span> from which z-scores can be
derived. Alternatively, you can obtain z-scores from <span class="math inline">\hat{\beta}</span> and p-values if you are provided
with those information. For example,</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sumstats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/univariate_regression.html">univariate_regression</a></span><span class="op">(</span><span class="va">X</span>, <span class="va">Y</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">z_scores</span> <span class="op">&lt;-</span> <span class="va">sumstats</span><span class="op">$</span><span class="va">betahat</span> <span class="op">/</span> <span class="va">sumstats</span><span class="op">$</span><span class="va">sebetahat</span></span>
<span><span class="fu"><a href="../reference/susie_plots.html">susie_plot</a></span><span class="op">(</span><span class="va">z_scores</span>, y <span class="op">=</span> <span class="st">"z"</span>, b<span class="op">=</span><span class="va">b</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-6-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<p>For this example, the correlation matrix can be computed directly
from data provided:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">R</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">X</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="fine-mapping-with-susier-using-summary-statistics">Fine-mapping with <code>susieR</code> using summary statistics<a class="anchor" aria-label="anchor" href="#fine-mapping-with-susier-using-summary-statistics"></a>
</h2>
<p>By default, SuSiE assumes at most 10 causal variables, with
<code>L = 10</code>, although we note that SuSiE is generally robust to
the choice of <code>L</code>.</p>
<p>Since the individual-level data is available for us here, we can
easily compute the “in-sample LD” matrix, as well as the variance of
<span class="math inline">y</span>, which is 7.8424. (By “in-sample”, we
means the LD was computed from the exact same matrix <code>X</code> that
was used to obtain the other statistics.) When we fit SuSiE regression
with summary statistics, <span class="math inline">\hat{\beta}</span>,
<span class="math inline">SE(\hat{\beta})</span>, <span class="math inline">R</span>, <span class="math inline">n</span>, and
var_y these are also the <em>sufficient statistics.</em> With an
in-sample LD, we can also estimate the residual variance using these
sufficient statistics. (Note that if the covariate effects are removed
from the genotypes in the univariate regression, it is recommended that
the “in-sample” LD matrix also be computed from the genotypes after the
covariate effects have been removed.)</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fitted_rss1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/susie_rss.html">susie_rss</a></span><span class="op">(</span>bhat <span class="op">=</span> <span class="va">sumstats</span><span class="op">$</span><span class="va">betahat</span>, shat <span class="op">=</span> <span class="va">sumstats</span><span class="op">$</span><span class="va">sebetahat</span>, n <span class="op">=</span> <span class="va">n</span>, R <span class="op">=</span> <span class="va">R</span>, var_y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">var</a></span><span class="op">(</span><span class="va">Y</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span>, L <span class="op">=</span> <span class="fl">10</span>,</span>
<span>                         estimate_residual_variance <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># <span style="color: #BB00BB; font-weight: bold; text-decoration: underline;">HINT:</span> For estimate_residual_variance = TRUE, please check that R is the "in-sample" LD matrix; that is, the correlation matrix obtained using the exact same data matrix X that was used for the other summary statistics. Also note, when covariates are included in the univariate regressions that produced the summary statistics, also consider removing these effects from X before computing R.</span></span></code></pre></div>
<p>Using <code>summary</code>, we can examine the posterior inclusion
probability (PIP) for each variable, and the 95% credible sets:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">fitted_rss1</span><span class="op">)</span><span class="op">$</span><span class="va">cs</span></span>
<span><span class="co">#   cs cs_log10bf cs_avg_r2 cs_min_r2</span></span>
<span><span class="co"># 1  2   4.033886 1.0000000 1.0000000</span></span>
<span><span class="co"># 2  1   6.744092 0.9634847 0.9634847</span></span>
<span><span class="co"># 3  3   3.461476 0.9293299 0.7545197</span></span>
<span><span class="co">#                                                                                                      variable</span></span>
<span><span class="co"># 1                                                                                                         653</span></span>
<span><span class="co"># 2                                                                                                     773,777</span></span>
<span><span class="co"># 3 362,365,372,373,374,379,381,383,384,386,387,388,389,391,392,396,397,398,399,400,401,403,404,405,407,408,415</span></span></code></pre></div>
<p>The three causal signals have been captured by the three CSs. Note
the third CS contains many variables, including the true causal variable
403.</p>
<p>We can also plot the posterior inclusion probabilities (PIPs):</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/susie_plots.html">susie_plot</a></span><span class="op">(</span><span class="va">fitted_rss1</span>, y<span class="op">=</span><span class="st">"PIP"</span>, b<span class="op">=</span><span class="va">b</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-10-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<p>The true causal variables are shown in red. The 95% CSs are shown by
three different colours (green, purple, blue).</p>
<p>Note this result is <em>exactly the same</em> as running
<code>susie</code> on the original individual-level data:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fitted</span> <span class="op">=</span> <span class="fu"><a href="../reference/susie.html">susie</a></span><span class="op">(</span><span class="va">X</span>, <span class="va">Y</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>, L <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/all.equal.html" class="external-link">all.equal</a></span><span class="op">(</span><span class="va">fitted</span><span class="op">$</span><span class="va">pip</span>, <span class="va">fitted_rss1</span><span class="op">$</span><span class="va">pip</span><span class="op">)</span></span>
<span><span class="co"># [1] TRUE</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/all.equal.html" class="external-link">all.equal</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">fitted</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">fitted_rss1</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co"># [1] TRUE</span></span></code></pre></div>
<p>If, on the other hand, the variance of <span class="math inline">y</span> is unknown, we fit can SuSiE regression
with summary statistics, <span class="math inline">\hat{\beta}</span>,
<span class="math inline">SE(\hat{\beta})</span>, <span class="math inline">R</span> and <span class="math inline">n</span> (or
<em>z</em>-scores, <span class="math inline">R</span> and <span class="math inline">n</span>). The outputted effect estimates are now on
the standardized <span class="math inline">X</span> and <span class="math inline">y</span> scale. Still, we can estimate the residual
variance because we have the in-sample LD matrix:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fitted_rss2</span> <span class="op">=</span> <span class="fu"><a href="../reference/susie_rss.html">susie_rss</a></span><span class="op">(</span>z <span class="op">=</span> <span class="va">z_scores</span>, R <span class="op">=</span> <span class="va">R</span>, n <span class="op">=</span> <span class="va">n</span>, L <span class="op">=</span> <span class="fl">10</span>,</span>
<span>                        estimate_residual_variance <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># <span style="color: #BB00BB; font-weight: bold; text-decoration: underline;">HINT:</span> For estimate_residual_variance = TRUE, please check that R is the "in-sample" LD matrix; that is, the correlation matrix obtained using the exact same data matrix X that was used for the other summary statistics. Also note, when covariates are included in the univariate regressions that produced the summary statistics, also consider removing these effects from X before computing R.</span></span></code></pre></div>
<p>The result is same as if we had run <code>susie</code> on the
individual-level data, but the output effect estimates are on different
scale:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/all.equal.html" class="external-link">all.equal</a></span><span class="op">(</span><span class="va">fitted</span><span class="op">$</span><span class="va">pip</span>, <span class="va">fitted_rss2</span><span class="op">$</span><span class="va">pip</span><span class="op">)</span></span>
<span><span class="co"># [1] TRUE</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">fitted</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">fitted_rss2</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span>, xlab <span class="op">=</span> <span class="st">'effects from SuSiE'</span>, ylab <span class="op">=</span> <span class="st">'effects from SuSiE-RSS'</span>, xlim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>,<span class="fl">1</span><span class="op">)</span>, ylim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.3</span>,<span class="fl">0.3</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-13-1.png" alt="&amp;nbsp;" width="420"><p class="caption">
 
</p>
</div>
<p>Specifically, without the variance of <span class="math inline">y</span>, these estimates are same as if we had
applied SuSiE to a standardized <span class="math inline">X</span> and
<span class="math inline">y</span>; that is, as if <span class="math inline">y</span> and the columns of <span class="math inline">X</span> had been normalized so that <span class="math inline">y</span> and each column of <span class="math inline">X</span> had a standard deviation of 1.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fitted_standardize</span> <span class="op">=</span> <span class="fu"><a href="../reference/susie.html">susie</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="va">X</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="va">Y</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span>, L <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/all.equal.html" class="external-link">all.equal</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">fitted_standardize</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">fitted_rss2</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co"># [1] TRUE</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="fine-mapping-with-susier-using-ld-matrix-from-reference-panel">Fine-mapping with <code>susieR</code> using LD matrix from reference
panel<a class="anchor" aria-label="anchor" href="#fine-mapping-with-susier-using-ld-matrix-from-reference-panel"></a>
</h2>
<p>When the original genotypes are not available, one may use a separate
reference panel to estimate the LD matrix.</p>
<p>To illustrate this, we randomly generated 500 samples from <span class="math inline">N(0,R)</span> and treated them as reference panel
genotype matrix <code>X_ref</code>.</p>
<p>We fit the SuSiE regression using out-of sample LD matrix. The
residual variance is fixed at 1 because estimating residual variance
sometimes produces very inaccurate estimates with out-of-sample LD
matrix. The output effect estimates are on the standardized <span class="math inline">X</span> and <span class="math inline">y</span>
scale.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fitted_rss3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/susie_rss.html">susie_rss</a></span><span class="op">(</span><span class="va">z_scores</span>, <span class="va">R_ref</span>, n<span class="op">=</span><span class="va">n</span>, L <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/susie_plots.html">susie_plot</a></span><span class="op">(</span><span class="va">fitted_rss3</span>, y<span class="op">=</span><span class="st">"PIP"</span>, b<span class="op">=</span><span class="va">b</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-17-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<p>In this particular example, the SuSiE result with out-of-sample LD is
very similar to using the in-sample LD matrix because the LD matrices
are quite similar.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fitted_rss1</span><span class="op">$</span><span class="va">pip</span>, <span class="va">fitted_rss3</span><span class="op">$</span><span class="va">pip</span>, ylim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, xlab<span class="op">=</span><span class="st">'SuSiE PIP'</span>, ylab<span class="op">=</span><span class="st">'SuSiE-RSS PIP'</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-18-1.png" alt="&amp;nbsp;" width="420"><p class="caption">
 
</p>
</div>
<p>In some rare cases, the sample size <span class="math inline">n</span> is unknown. When the sample size is not
provided as input to <code>susie_rss</code>, this is in effect assuming
the sample size is infinity and all the effects are small, and the
estimated PVE for each variant will be close to zero. The outputted
effect estimates are on the “noncentrality parameter” scale.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fitted_rss4</span> <span class="op">=</span> <span class="fu"><a href="../reference/susie_rss.html">susie_rss</a></span><span class="op">(</span><span class="va">z_scores</span>, <span class="va">R_ref</span>, L <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="co"># <span style="color: #BB0000; font-weight: bold; text-decoration: underline;">WARNING:</span> Providing the sample size (n), or even a rough estimate of n, is highly recommended. Without n, the implicit assumption is n is large (Inf) and the effect sizes are small (close to zero).</span></span>
<span><span class="fu"><a href="../reference/susie_plots.html">susie_plot</a></span><span class="op">(</span><span class="va">fitted_rss4</span>, y<span class="op">=</span><span class="st">"PIP"</span>, b<span class="op">=</span><span class="va">b</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-19-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>Here are some details about the computing environment, including the
versions of R, and the R packages, used to generate these results.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co"># R version 4.4.3 (2025-02-28)</span></span>
<span><span class="co"># Platform: x86_64-conda-linux-gnu</span></span>
<span><span class="co"># Running under: Linux Mint 22</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># Matrix products: default</span></span>
<span><span class="co"># BLAS/LAPACK: /home/gw/.pixi/envs/r-base/lib/libopenblasp-r0.3.30.so;  LAPACK version 3.12.0</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># locale:</span></span>
<span><span class="co">#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co"># [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co"># </span></span>
<span><span class="co"># time zone: America/New_York</span></span>
<span><span class="co"># tzcode source: system (glibc)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># attached base packages:</span></span>
<span><span class="co"># [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co"># </span></span>
<span><span class="co"># other attached packages:</span></span>
<span><span class="co"># [1] susieR_0.14.21</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># loaded via a namespace (and not attached):</span></span>
<span><span class="co">#  [1] Matrix_1.7-4       gtable_0.3.6       jsonlite_2.0.0     dplyr_1.1.4       </span></span>
<span><span class="co">#  [5] compiler_4.4.3     crayon_1.5.3       Rcpp_1.1.0         tidyselect_1.2.1  </span></span>
<span><span class="co">#  [9] parallel_4.4.3     jquerylib_0.1.4    systemfonts_1.2.3  scales_1.4.0      </span></span>
<span><span class="co"># [13] textshaping_1.0.3  yaml_2.3.10        fastmap_1.2.0      lattice_0.22-7    </span></span>
<span><span class="co"># [17] plyr_1.8.9         ggplot2_4.0.0      R6_2.6.1           generics_0.1.4    </span></span>
<span><span class="co"># [21] mixsqp_0.3-54      knitr_1.50         htmlwidgets_1.6.4  tibble_3.3.0      </span></span>
<span><span class="co"># [25] desc_1.4.3         RcppZiggurat_0.1.8 bslib_0.9.0        pillar_1.11.1     </span></span>
<span><span class="co"># [29] RColorBrewer_1.1-3 rlang_1.1.6        reshape_0.8.10     cachem_1.1.0      </span></span>
<span><span class="co"># [33] xfun_0.53          fs_1.6.6           sass_0.4.10        S7_0.2.0          </span></span>
<span><span class="co"># [37] RcppParallel_5.1.9 cli_3.6.5          pkgdown_2.1.3      magrittr_2.0.4    </span></span>
<span><span class="co"># [41] digest_0.6.37      grid_4.4.3         irlba_2.3.5.1      lifecycle_1.0.4   </span></span>
<span><span class="co"># [45] vctrs_0.6.5        Rfast_2.1.0        evaluate_1.0.5     glue_1.8.0        </span></span>
<span><span class="co"># [49] farver_2.1.2       ragg_1.5.0         rmarkdown_2.29     matrixStats_1.5.0 </span></span>
<span><span class="co"># [53] tools_4.4.3        pkgconfig_2.0.3    htmltools_0.5.8.1</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Gao Wang, Yuxin Zou, Kaiqian Zhang, William R.P. Denault, Peter Carbonetto, Matthew Stephens.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
