<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Diagnostic for fine-mapping with summary statistics â€¢ susieR</title>
<!-- katex math --><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.16.11/dist/katex.min.css" integrity="sha384-nB0miv6/jRmo5UMMR1wu3Gz6NLsoTkbqJghGIsx//Rlm+ZU03BU6SQNC66uf4l5+" crossorigin="anonymous">
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.11/dist/katex.min.js" integrity="sha384-7zkQWkzuo3B5mTepMUcHkMB5jZaolc2xDwL6VFqjFALcbeS9Ggm/Yr2r3Dy4lfFg" crossorigin="anonymous"></script><script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.11/dist/contrib/auto-render.min.js" integrity="sha384-43gviWU0YVjaDtb/GhzOouOXtZMP/7XUzwPTstBeZFe/+rCMvRwr4yROQP43s0Xk" crossorigin="anonymous" onload="renderMathInElement(document.body);"></script><script src="../katex-auto.js"></script><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Roboto-0.4.10/font.css" rel="stylesheet">
<link href="../deps/JetBrains_Mono-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Roboto_Slab-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Diagnostic for fine-mapping with summary statistics">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">susieR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.14.21</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/finemapping.html" aria-label="Get started with susieR"><span class="fa fa-play-circle"></span> Get Started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html" aria-label="Function reference"><span class="fa fa-file-code"></span> Functions</a></li>
<li class="active nav-item"><a class="nav-link" href="../articles/index.html" aria-label="Articles and vignettes"><span class="fa fa-book-reader"></span> Vignettes</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html" aria-label="Package news"><span class="fa fa-newspaper"></span> News</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/stephenslab/susieR" aria-label="View source on GitHub"><span class="fa fab fa-github"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Diagnostic for fine-mapping with summary statistics</h1>
                        <h4 data-toc-skip class="author">Yuxin Zou</h4>
            
            <h4 data-toc-skip class="date">2025-09-24</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/stephenslab/susieR/blob/HEAD/vignettes/susierss_diagnostic.Rmd" class="external-link"><code>vignettes/susierss_diagnostic.Rmd</code></a></small>
      <div class="d-none name"><code>susierss_diagnostic.Rmd</code></div>
    </div>

    
    
<p>This vignette demonstrates the use of the diagnostic plot for
assessing consistency of the summary statistics and the reference LD
matrix.</p>
<p>The <code>susie_rss</code> assumes the LD matrix accurately estimate
the correlations among SNPs from the original GWAS genotype data.
Typically, the LD matrix comes from some public database of genotypes in
a suitable reference population. The inaccurate LD information leads to
unreliable fine-mapping result.</p>
<p>The diagnostic for consistency between summary statistics and
refenrence LD matrix is based on the RSS model under the null with
regularized LD matrix. <span class="math display">
\hat{z} | R, \lambda \sim N(0, (1-\lambda)R + \lambda I),
0&lt;\lambda&lt;1
</span> The parameter <span class="math inline">\lambda</span> is
estimated by maximum likelihood. A larger <span class="math inline">\lambda</span> means a greater inconsistency between
summary statistics and the LD matrix. The expected z score is computed
for each SNP, <span class="math inline">E(\hat{z}_j |
\hat{z}_{-j})</span>, and plotted against the observed z scores.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/stephenslab/susieR" class="external-link">susieR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://jeroen.r-universe.dev/curl" class="external-link">curl</a></span><span class="op">)</span></span>
<span><span class="co"># Using libcurl 8.14.1 with OpenSSL/3.5.3</span></span></code></pre></div>
<div class="section level2">
<h2 id="ld-information-from-the-original-genotype-data">LD information from the original genotype data<a class="anchor" aria-label="anchor" href="#ld-information-from-the-original-genotype-data"></a>
</h2>
<p>We demonstrate the diagnostic plot in a simple case, the LD matrix is
estimated from the original genotype data. In this case, we expect the
diagnostic plot to confirm that the LD matrix is consistent with the z
scores.</p>
<p>We use the same simulated data as in <a href="finemapping.html">fine
mapping vignette</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"N3finemapping"</span><span class="op">)</span></span>
<span><span class="va">n</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">N3finemapping</span><span class="op">$</span><span class="va">X</span><span class="op">)</span></span>
<span><span class="va">b</span> <span class="op">=</span> <span class="va">N3finemapping</span><span class="op">$</span><span class="va">true_coef</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">sumstats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/univariate_regression.html">univariate_regression</a></span><span class="op">(</span><span class="va">N3finemapping</span><span class="op">$</span><span class="va">X</span>, <span class="va">N3finemapping</span><span class="op">$</span><span class="va">Y</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">z_scores</span> <span class="op">&lt;-</span> <span class="va">sumstats</span><span class="op">$</span><span class="va">betahat</span> <span class="op">/</span> <span class="va">sumstats</span><span class="op">$</span><span class="va">sebetahat</span></span>
<span><span class="va">Rin</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">N3finemapping</span><span class="op">$</span><span class="va">X</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/attr.html" class="external-link">attr</a></span><span class="op">(</span><span class="va">Rin</span>, <span class="st">"eigen"</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/eigen.html" class="external-link">eigen</a></span><span class="op">(</span><span class="va">Rin</span>, symmetric <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/susie_plots.html">susie_plot</a></span><span class="op">(</span><span class="va">z_scores</span>, y <span class="op">=</span> <span class="st">"z"</span>, b<span class="op">=</span><span class="va">b</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="susierss_diagnostic_files/figure-html/unnamed-chunk-2-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
Â 
</p>
</div>
<p>The estimated <span class="math inline">\lambda</span> is</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lambda</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimate_s_rss.html">estimate_s_rss</a></span><span class="op">(</span><span class="va">z_scores</span>, <span class="va">Rin</span>, n<span class="op">=</span><span class="va">n</span><span class="op">)</span></span>
<span><span class="va">lambda</span></span>
<span><span class="co"># [1] 6.474096e-09</span></span></code></pre></div>
<p>The plot for the observed z scores vs the expected z scores is</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">condz_in</span> <span class="op">=</span> <span class="fu"><a href="../reference/kriging_rss.html">kriging_rss</a></span><span class="op">(</span><span class="va">z_scores</span>, <span class="va">Rin</span>, n<span class="op">=</span><span class="va">n</span><span class="op">)</span></span>
<span><span class="co"># Warning: `aes_string()` was deprecated in ggplot2 3.0.0.</span></span>
<span><span class="co"># <span style="color: #00BBBB;">â„¹</span> Please use tidy evaluation idioms with `aes()`.</span></span>
<span><span class="co"># <span style="color: #00BBBB;">â„¹</span> See also `vignette("ggplot2-in-packages")` for more information.</span></span>
<span><span class="co"># <span style="color: #00BBBB;">â„¹</span> The deprecated feature was likely used in the <span style="color: #0000BB;">susieR</span> package.</span></span>
<span><span class="co">#   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/stephenslab/susieR/issues&gt;</span>.</span></span>
<span><span class="co"># <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co"># <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co"># <span style="color: #555555;">generated.</span></span></span>
<span><span class="va">condz_in</span><span class="op">$</span><span class="va">plot</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="susierss_diagnostic_files/figure-html/unnamed-chunk-4-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
Â 
</p>
</div>
<p>Summary of SuSiE Credible Sets:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/susie_rss.html">susie_rss</a></span><span class="op">(</span><span class="va">z_scores</span>, <span class="va">Rin</span>, n<span class="op">=</span><span class="va">n</span>, estimate_residual_variance <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># <span style="color: #BB00BB; font-weight: bold; text-decoration: underline;">HINT:</span> For estimate_residual_variance = TRUE, please check that R is the "in-sample" LD matrix; that is, the correlation matrix obtained using the exact same data matrix X that was used for the other summary statistics. Also note, when covariates are included in the univariate regressions that produced the summary statistics, also consider removing these effects from X before computing R.</span></span>
<span><span class="fu"><a href="../reference/susie_plots.html">susie_plot</a></span><span class="op">(</span><span class="va">fit</span>,y <span class="op">=</span> <span class="st">"PIP"</span>, b<span class="op">=</span><span class="va">b</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="susierss_diagnostic_files/figure-html/unnamed-chunk-5-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
Â 
</p>
</div>
</div>
<div class="section level2">
<h2 id="ld-information-from-the-reference-panel">LD information from the reference panel<a class="anchor" aria-label="anchor" href="#ld-information-from-the-reference-panel"></a>
</h2>
<p>We use another simulated data where the LD matrix is estimated from a
reference panel. In this example data set, there is one association
signal in the simulated data (red point), and there is one SNP with
mismatched reference and alternative allele between summary statistics
and the reference panel (yellow point).</p>
<p><strong>Note:</strong> In some versions of <a href="https://www.cog-genomics.org/plink" class="external-link">PLINK</a>, these mismatches
can occur when <a href="https://github.com/stephenslab/susieR/issues/148" class="external-link">PLINK
automatically flips the alleles to make the minor allele be the effect
allele</a>, leading to different allele encodings in the z scores and LD
matrix. Adding the flag <code>--keep-allele-order</code> will disable
this behaviour in PLINK.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".RData"</span><span class="op">)</span></span>
<span><span class="va">data_url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"https://raw.githubusercontent.com/stephenslab/susieR/"</span>,</span>
<span>                   <span class="st">"master/inst/datafiles/SummaryConsistency1k.RData"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://jeroen.r-universe.dev/curl/reference/curl_download.html" class="external-link">curl_download</a></span><span class="op">(</span><span class="va">data_url</span>,<span class="va">data_file</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">data_file</span><span class="op">)</span></span>
<span><span class="va">zflip</span> <span class="op">=</span> <span class="va">SummaryConsistency</span><span class="op">$</span><span class="va">z</span></span>
<span><span class="va">ld</span> <span class="op">=</span> <span class="va">SummaryConsistency</span><span class="op">$</span><span class="va">ldref</span></span>
<span><span class="va">n</span><span class="op">=</span><span class="fl">10000</span></span>
<span><span class="va">b</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">zflip</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">b</span><span class="op">[</span><span class="va">SummaryConsistency</span><span class="op">$</span><span class="va">signal_id</span><span class="op">]</span> <span class="op">=</span> <span class="va">zflip</span><span class="op">[</span><span class="va">SummaryConsistency</span><span class="op">$</span><span class="va">signal_id</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">zflip</span>, pch <span class="op">=</span> <span class="fl">16</span>, col <span class="op">=</span> <span class="st">"#767676"</span>, main <span class="op">=</span> <span class="st">"Marginal Associations"</span>, </span>
<span>     xlab<span class="op">=</span><span class="st">"SNP"</span>, ylab <span class="op">=</span> <span class="st">"z scores"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="va">SummaryConsistency</span><span class="op">$</span><span class="va">signal_id</span>, <span class="va">zflip</span><span class="op">[</span><span class="va">SummaryConsistency</span><span class="op">$</span><span class="va">signal_id</span><span class="op">]</span>, col<span class="op">=</span><span class="fl">2</span>, pch<span class="op">=</span><span class="fl">16</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="va">SummaryConsistency</span><span class="op">$</span><span class="va">flip_id</span>, <span class="va">zflip</span><span class="op">[</span><span class="va">SummaryConsistency</span><span class="op">$</span><span class="va">flip_id</span><span class="op">]</span>, col<span class="op">=</span><span class="fl">7</span>, pch<span class="op">=</span><span class="fl">16</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="susierss_diagnostic_files/figure-html/unnamed-chunk-6-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
Â 
</p>
</div>
<p>Using the data with misaligned allele, SuSiE-RSS identifies a true
positive CS containing the true effect SNP; and a false positive CS that
incorrectly contains the mismatched SNP.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit</span> <span class="op">=</span> <span class="fu"><a href="../reference/susie_rss.html">susie_rss</a></span><span class="op">(</span><span class="va">zflip</span>, <span class="va">ld</span>, n<span class="op">=</span><span class="va">n</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/susie_plots.html">susie_plot</a></span><span class="op">(</span><span class="va">fit</span>, y<span class="op">=</span><span class="st">'PIP'</span>, b<span class="op">=</span><span class="va">b</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="va">SummaryConsistency</span><span class="op">$</span><span class="va">flip_id</span>, <span class="va">fit</span><span class="op">$</span><span class="va">pip</span><span class="op">[</span><span class="va">SummaryConsistency</span><span class="op">$</span><span class="va">flip_id</span><span class="op">]</span>, col<span class="op">=</span><span class="fl">7</span>, pch<span class="op">=</span><span class="fl">16</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="susierss_diagnostic_files/figure-html/unnamed-chunk-7-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
Â 
</p>
</div>
<p>The estimated <span class="math inline">\lambda</span> is</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lambda</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimate_s_rss.html">estimate_s_rss</a></span><span class="op">(</span><span class="va">zflip</span>, <span class="va">ld</span>, n<span class="op">=</span><span class="va">n</span><span class="op">)</span></span>
<span><span class="va">lambda</span></span>
<span><span class="co"># [1] 0.005005011</span></span></code></pre></div>
<p>In the diagnostic plot, the mismatched SNP shows the largest
difference between observed and expected z-scores, and therefore appears
furthest away from the diagonal.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">condz</span> <span class="op">=</span> <span class="fu"><a href="../reference/kriging_rss.html">kriging_rss</a></span><span class="op">(</span><span class="va">zflip</span>, <span class="va">ld</span>, n<span class="op">=</span><span class="va">n</span><span class="op">)</span></span>
<span><span class="va">condz</span><span class="op">$</span><span class="va">plot</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="susierss_diagnostic_files/figure-html/unnamed-chunk-9-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
Â 
</p>
</div>
<p>After fixing the allele encoding, SuSiE-RSS identifies a single true
positive CS containing the true-effect SNP, and the formerly mismatched
SNP is (correctly) not included in a CS.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">z</span> <span class="op">=</span> <span class="va">zflip</span></span>
<span><span class="va">z</span><span class="op">[</span><span class="va">SummaryConsistency</span><span class="op">$</span><span class="va">flip_id</span><span class="op">]</span> <span class="op">=</span> <span class="op">-</span><span class="va">z</span><span class="op">[</span><span class="va">SummaryConsistency</span><span class="op">$</span><span class="va">flip_id</span><span class="op">]</span></span>
<span><span class="va">fit</span> <span class="op">=</span> <span class="fu"><a href="../reference/susie_rss.html">susie_rss</a></span><span class="op">(</span><span class="va">z</span>, <span class="va">ld</span>, n<span class="op">=</span><span class="va">n</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/susie_plots.html">susie_plot</a></span><span class="op">(</span><span class="va">fit</span>, y<span class="op">=</span><span class="st">'PIP'</span>, b<span class="op">=</span><span class="va">b</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="susierss_diagnostic_files/figure-html/unnamed-chunk-10-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
Â 
</p>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>Here are some details about the computing environment, including the
versions of R, and the R packages, used to generate these results.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co"># R version 4.4.3 (2025-02-28)</span></span>
<span><span class="co"># Platform: x86_64-conda-linux-gnu</span></span>
<span><span class="co"># Running under: Linux Mint 22</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># Matrix products: default</span></span>
<span><span class="co"># BLAS/LAPACK: /home/gw/.pixi/envs/r-base/lib/libopenblasp-r0.3.30.so;  LAPACK version 3.12.0</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># locale:</span></span>
<span><span class="co">#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co"># [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co"># </span></span>
<span><span class="co"># time zone: America/New_York</span></span>
<span><span class="co"># tzcode source: system (glibc)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># attached base packages:</span></span>
<span><span class="co"># [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co"># </span></span>
<span><span class="co"># other attached packages:</span></span>
<span><span class="co"># [1] curl_7.0.0     susieR_0.14.21</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># loaded via a namespace (and not attached):</span></span>
<span><span class="co">#  [1] Matrix_1.7-4       gtable_0.3.6       jsonlite_2.0.0     dplyr_1.1.4       </span></span>
<span><span class="co">#  [5] compiler_4.4.3     crayon_1.5.3       Rcpp_1.1.0         tidyselect_1.2.1  </span></span>
<span><span class="co">#  [9] parallel_4.4.3     jquerylib_0.1.4    systemfonts_1.2.3  scales_1.4.0      </span></span>
<span><span class="co"># [13] textshaping_1.0.3  yaml_2.3.10        fastmap_1.2.0      lattice_0.22-7    </span></span>
<span><span class="co"># [17] plyr_1.8.9         ggplot2_4.0.0      R6_2.6.1           labeling_0.4.3    </span></span>
<span><span class="co"># [21] generics_0.1.4     mixsqp_0.3-54      knitr_1.50         htmlwidgets_1.6.4 </span></span>
<span><span class="co"># [25] tibble_3.3.0       desc_1.4.3         RcppZiggurat_0.1.8 bslib_0.9.0       </span></span>
<span><span class="co"># [29] pillar_1.11.1      RColorBrewer_1.1-3 rlang_1.1.6        reshape_0.8.10    </span></span>
<span><span class="co"># [33] cachem_1.1.0       xfun_0.53          fs_1.6.6           sass_0.4.10       </span></span>
<span><span class="co"># [37] S7_0.2.0           RcppParallel_5.1.9 cli_3.6.5          withr_3.0.2       </span></span>
<span><span class="co"># [41] pkgdown_2.1.3      magrittr_2.0.4     digest_0.6.37      grid_4.4.3        </span></span>
<span><span class="co"># [45] irlba_2.3.5.1      lifecycle_1.0.4    vctrs_0.6.5        Rfast_2.1.0       </span></span>
<span><span class="co"># [49] evaluate_1.0.5     glue_1.8.0         farver_2.1.2       ragg_1.5.0        </span></span>
<span><span class="co"># [53] rmarkdown_2.29     matrixStats_1.5.0  tools_4.4.3        pkgconfig_2.0.3   </span></span>
<span><span class="co"># [57] htmltools_0.5.8.1</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Gao Wang, Yuxin Zou, Kaiqian Zhang, William R.P. Denault, Peter Carbonetto, Matthew Stephens.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
